docs/start/framework/react/environment-functions.md

```diff
@@ -0,0 +1,105 @@
+---
+id: environment-functions
+title: Environment Functions
+---
+
+## What Are Environment Functions?
+
+Environment functions are utilities designed to define and control function execution based on the runtime environment—whether the code is running on the client or the server. These utilities help ensure that environment-specific logic is executed safely and intentionally, preventing runtime errors and improving maintainability in fullstack or isomorphic applications.
+
+Start provide three core environment functions:
+
+- `createIsomorphicFn`: Compose a single function that adapts to both client and server environments.
+- `serverOnly`: Ensures a function can only run on the server.
+- `clientOnly`: Ensures a function can only run on the client.
+
+---
+
+## Isomorphic Functions
+
+Use `createIsomorphicFn()` to define functions that behave differently depending on whether they are called on the client or the server. This is useful for safely sharing logic across environments while delegating environment-specific behavior to appropriate handlers. 
+
+> [!WARNING]
+> Unimplemented function results in a no-op function.
```

Comment:
Maybe instead of a warning, this could be a small heading like "What is a no-op?", that'd show the user the type of a no-op.

Perhaps something like this:

```ts
type NoOpFunction = () => undefined // add a comment here
```